### PR TITLE
Fix date format at multiple places

### PR DESCRIPTION
**Pull request by shalapatil**
Notion: https://www.notion.so/saeloun/The-date-format-is-incorrect-at-all-places-3f709da1a875466d8433b2031de9fe70?pvs=4

What: Fixes date format at backend as well on frontend side at following places 
- Invoice list, create, view, edit, send,
- Payment list, create
- Reports

Why:  One PR was already raised and merged which fixed date format for invoice and payment from backend side. 
But frontend had hardcoded the format and hence needed to fix it. 

Loom video: https://www.loom.com/share/ab44eebbbbae478d8a6317b6ff71282f

### CODE REVIEW

File: app/views/internal_api/v1/payments/new.json.jbuilder [hunk 1]
```diff
@@ -11,3 +11,7 @@ json.invoices invoices do |invoice|
   json.amount invoice.amount
   json.status invoice.status
 end
+
+json.company do
+  json.date_format company.date_format
+end
```
Comment (vipulnsward) [inline app/views/internal_api/v1/payments/new.json.jbuilder:17]: Missing EOF

File: app/javascript/src/components/payments/Modals/AddManualEntry.tsx [hunk 2]
```diff
@@ -149,7 +150,7 @@ const AddManualEntry = ({
             {props.data.amount}
           </h1>
           <h3 className="pt-1 text-sm font-normal leading-5 text-miru-dark-purple-400">
-            {props.data.invoiceDate}
+            {dayjs(props.data.invoiceDate).format(dateFormat)}
```
Comment (vipulnsward) [inline app/javascript/src/components/payments/Modals/AddManualEntry.tsx:153]: Destructure `data` variable from props and move to fetching `invoiceDate`, `invoiceNumber ` from above this instead.

File: app/javascript/src/components/payments/index.tsx [hunk 3]
```diff
@@ -16,13 +16,15 @@ const Payments = () => {
...
   const [invoiceList, setInvoiceList] = useState<any>([]);
+  const [dateFormat, setDateFormat] = useState<any>("");
   const [baseCurrency, setBaseCurrency] = useState<any>("");
   const { isDesktop } = useUserContext();
   const fetchInvoiceList = async () => {
     try {
       const res = await payment.getInvoiceList();
       const sanitzed = await unmapPayment(res.data);
       setInvoiceList(sanitzed);
+      setDateFormat(res.data.company.dateFormat);
```
Comment (vipulnsward) [inline app/javascript/src/components/payments/index.tsx:27]: Same here.

Comment (shalapatil) [reply app/javascript/src/components/payments/index.tsx]: done

File: app/javascript/src/components/payments/Modals/AddManualEntry.tsx (hunk 2 above)
Comment (shalapatil) [reply app/javascript/src/components/payments/Modals/AddManualEntry.tsx]: done

File: app/views/internal_api/v1/payments/new.json.jbuilder (hunk 1 above)
Comment (shalapatil) [reply app/views/internal_api/v1/payments/new.json.jbuilder]: done

**Review by vipulnsward — APPROVED** [review]

**Pull request merged**